editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -2320,28 +2320,33 @@ bool CanvasItemEditor::_gui_input_select(const Ref<InputEvent> &p_event) {
 			}
 		}
 
-		if (b.is_valid() && b->is_pressed() && b->get_button_index() == MouseButton::RIGHT) {
-			add_node_menu->clear();
-			add_node_menu->add_icon_item(get_editor_theme_icon(SNAME("Add")), TTR("Add Node Here..."), ADD_NODE);
-			add_node_menu->add_icon_item(get_editor_theme_icon(SNAME("Instance")), TTR("Instantiate Scene Here..."), ADD_INSTANCE);
-			for (Node *node : SceneTreeDock::get_singleton()->get_node_clipboard()) {
-				if (Object::cast_to<CanvasItem>(node)) {
-					add_node_menu->add_icon_item(get_editor_theme_icon(SNAME("ActionPaste")), TTR("Paste Node(s) Here"), ADD_PASTE);
-					break;
+		if (b.is_valid() && b->get_button_index() == MouseButton::RIGHT) {
+			if (b->is_pressed()) {
+				right_click_origin = b->get_position();
+				return true;
+			} else if ((b->get_position() - right_click_origin).length_squared() <= 64) {
```

Comment:
The distance threshold seems pretty high. Also, I'd move it to a constant to avoid using a magic number.